### ISSUE TITLE
aws_rds_cluster_parameter_group does not generate a unique name when specifying name_prefix

### ISSUE THREAD

**Issue by majormoses** [terraform-providers/terraform-provider-aws#1739]
### Terraform Version
```
$ terraform --version
Terraform v0.9.11

Your version of Terraform is out of date! The latest version
is 0.10.6. You can update by downloading from www.terraform.io
```

### Affected Resource(s)
Please list the resources as a list, for example:
- aws_rds_cluster_parameter_group

### Terraform Configuration Files
```hcl
####
# RDS Cluster Paramater Group

resource "aws_rds_cluster_parameter_group" "fa-arora" {
  name_prefix = "${var.env}-aurora-params"
  family      = "aurora5.6"
  description = "RDS ${var.env} cluster parameter group"

  parameter {
    name  = "character_set_server"
    value = "utf8"
  }

  parameter {
    name  = "character_set_client"
    value = "utf8"
  }

}
# End of RDS Cluster Paramater Group
####
```

### Debug Output
https://gist.github.com/majormoses/f373836f0219ae37a0079f3938868ce9

### Expected Behavior
Terraform should have computed a unique name (using the prefix) and submitted it with the request.

### Actual Behavior
It appears that the plan believes it should compute the value but either does and does not submit it or does not compute it. This results in a 400 because the body did not match the requirements:
```
* aws_rds_cluster_parameter_group.fa-arora: Error modifying DB Cluster Parameter Group: InvalidParameterValue: The parameter DBClusterParameterGroupName must be provided and must not be blank.
	status code: 400, request id: 40a2e32e-a094-11e7-a4f7-21baf61bfe2b
```

### Steps to Reproduce
Please list the steps required to reproduce the issue, for example:
1. `terraform plan`
1. `terraform apply`

### Important Factoids
None

### References
Are there any other GitHub issues (open or closed) or Pull Requests that should be linked here?
Not that I saw

**Comment by majormoses** [comment]
Commenting out `name_prefix` and specifying the `name` attribute works as expected.

**Comment by spanktar** [comment]
0.10.7 confirmed. name_prefix combined with create_before_destroy.

**Comment by richid** [comment]
Confirmed in `0.11.5` as well.

Here is a little workaround I'm successfully using:
```
name = "${var.name}-${lower(replace(timestamp(), ":", ""))}"
```
Not as elegant as the `name_prefix` but it gets the job done.

**Comment by bflad** [comment]
The fix for the `aws_rds_cluster_parameter_group` resource not correctly applying during resource creation with a generated name or with the `name_prefix` argument while also including `parameter` configurations has been merged and will release with version 1.57.0 of the Terraform AWS provider, likely later today. 👍

**Comment by bflad** [comment]
This has been released in [version 1.57.0 of the AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#1570-january-26-2019). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.
